perllib/FixMyStreet/Script/Inactive.pm

```diff
@@ -110,22 +128,39 @@ sub anonymize_reports {
         });
 
         # Remove alerts - could just delete, but of interest how many there were, perhaps?
-        FixMyStreet::DB->resultset('Alert')->search({
+        $problem->alerts->search({
             user_id => { '!=' => $self->anonymous_user->id },
-            alert_type => 'new_updates',
-            parameter => $problem->id,
         })->update({
             user_id => $self->anonymous_user->id,
             whendisabled => \'current_timestamp',
         });
     }
 }
 
+sub delete_reports {
+    my $self = shift;
+
+    my $problems = _relevant_reports($self->delete);
+
+    while (my $problem = $problems->next) {
+        say "Deleting associated data of problem #" . $problem->id if $self->verbose;
+        next if $self->dry_run;
+
+        $problem->comments->delete;
+        $problem->questionnaires->delete;
+        $problem->alerts->delete;
+    }
+    say "Deleting all matching problems" if $self->verbose;
+    return if $self->dry_run;
```

Comment:
The pattern of wrapping everything in a transaction which is committed or rolled back according to command line flag (see e.g. `bin/tfl/import-categories` and `FixMyStreet::DB::Factories`) might be worth considering to make it less likely one of these would be missed off.

Reply:
This script sends email also, though, so can't really do a per-whole-run transaction in case there was e.g. an email failure causing the script to die halfway through.

Reply:
An excellent point! Nevermind then :)